src/WebJobs.Script/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -60,6 +61,40 @@ public static bool IsLegacyBlobTriggerFunction(this FunctionMetadata metadata)
             return false;
         }
 
+        /// <summary>
+        /// Gets the function group for the specified <see cref="FunctionMetadata"/>.
+        /// </summary>
+        /// <param name="metadata">The function metadata.</param>
+        /// <returns>The function group for this metadata or <c>null</c> if no group specified.</returns>
+        public static string GetFunctionGroup(this FunctionMetadata metadata)
+        {
+            if (metadata == null)
+            {
+                throw new ArgumentNullException(nameof(metadata));
+            }
+
+            if (metadata.Properties.TryGetValue(FunctionGroupKey, out object group))
+            {
+                return group.ToString();
+            }
+
+            // Ensure http and durable triggers are grouped together.
```

Comment:
What about EventGrid BlobTriggers, and EventGridTrigger iteself? My understanding is that you're trying to group all functions that are http/webhook triggered together?

Reply:
I'll add a TODO for that and we can sync after the holidays. In general HTTP is not a big concern as scale controller is going to ignore everything HTTP related anyways. Unless this work here is going to be used to _drive_ scale controller knowing what is and isn't HTTP, then yes we would need to solve that.

Reply:
Added. Let me know if it looks good.